core/server/master/src/main/java/alluxio/master/file/DefaultFileSystemMaster.java

```diff
@@ -4843,6 +4811,64 @@ public void heartbeat() throws InterruptedException {
     public void close() {} // Nothing to clean up.
   }
 
+  private final class AsyncInodeStoreLoadTask implements Runnable {
+    @Override
+    public void run() {
+      int numToBePersisted = mInodeTree.getToBePersistedIds().size();
+      int actualToBePersisted = 0;
+      long start = System.currentTimeMillis();
+      LOG.info("[AE-2584] start fs master start time, numToBePersisted:{}", numToBePersisted);
+
+      // Rebuild the list of persist jobs (mPersistJobs) and map of pending persist requests
+      // (mPersistRequests)
+      long persistInitialIntervalMs =
+              Configuration.getMs(PropertyKey.MASTER_PERSISTENCE_INITIAL_INTERVAL_MS);
+      long persistMaxIntervalMs =
+              Configuration.getMs(PropertyKey.MASTER_PERSISTENCE_MAX_INTERVAL_MS);
+      long persistMaxWaitMs =
+              Configuration.getMs(PropertyKey.MASTER_PERSISTENCE_MAX_TOTAL_WAIT_TIME_MS);
+
+      for (Long id : mInodeTree.getToBePersistedIds()) {
+        Inode inode = mInodeStore.get(id).get();
+
+        if (inode.isDirectory()
+                || !inode.asFile().isCompleted() // When file is completed it is added to persist reqs
+                || inode.getPersistenceState() != PersistenceState.TO_BE_PERSISTED
+                || inode.asFile().getShouldPersistTime() == Constants.NO_AUTO_PERSIST) {
+          continue;
+        }
+        actualToBePersisted++;
+        InodeFile inodeFile = inode.asFile();
+        if (inodeFile.getPersistJobId() == Constants.PERSISTENCE_INVALID_JOB_ID) {
+          mPersistRequests.put(inodeFile.getId(),
+                  new alluxio.time.ExponentialTimer(
+                          persistInitialIntervalMs,
+                          persistMaxIntervalMs,
+                          getPersistenceWaitTime(inodeFile.getShouldPersistTime()),
+                          persistMaxWaitMs));
+        } else {
+          AlluxioURI path;
+          try {
+            path = mInodeTree.getPath(inodeFile);
+          } catch (FileDoesNotExistException e) {
+            LOG.error("Failed to determine path for inode with id {}", id, e);
+            continue;
+          }
+          addPersistJob(id, inodeFile.getPersistJobId(),
+                  getPersistenceWaitTime(inodeFile.getShouldPersistTime()),
+                  path, inodeFile.getTempUfsPath());
+        }
+      }
+      long ed = System.currentTimeMillis();
+      long cachehit = MetricsSystem.counter(MetricKey.MASTER_INODE_CACHE_HITS.getName()).getCount();
+      long cachemiss = MetricsSystem.counter(MetricKey.MASTER_INODE_CACHE_MISSES.getName()).getCount();
+      LOG.info("[AE-2584] mInodeTree tobePersistedIds num of inodes: {}," +
```

Comment:
Maybe add a bit more description here like
"Completed async loading of to be persisted job at startup"